mod/wikirate_source/set/type/source/source.rb

```diff
@@ -36,7 +36,7 @@ def indirect_contributor_search_args
 
 event :process_source_url, :before=>:process_subcards, :on=>:create do
 #, :when=>proc{    |c| Card::Env.params[:sourcebox] == 'true'  } do
-  
+  return if subcards["+File"] or subcards["+Text"]
```

Comment:
Thanks for fixing this.  I should explain the rationale here.  If you put the test in :when, then the event does not actually happen when the test fails.  This means that any event that is supposed to happen before ore after process_source_url will not get triggered.